src/status_im/subs/wallet/activities.cljs

```diff
@@ -1,26 +1,82 @@
 (ns status-im.subs.wallet.activities
   (:require
+    [legacy.status-im.utils.hex :as utils.hex]
+    [native-module.core :as native-module]
+    [quo.foundations.resources :as quo.resources]
+    [quo.foundations.resources]
     [re-frame.core :as rf]
     [status-im.contexts.wallet.common.activity-tab.constants :as constants]
-    [utils.datetime :as datetime]))
+    [utils.datetime :as datetime]
+    [utils.money :as money]))
+
+(def precision 6)
 
 (rf/reg-sub
  :wallet/all-activities
  :<- [:wallet]
  :-> :activities)
 
-(rf/reg-sub :wallet/activities-for-current-viewing-account
+(defn- activity-amount
+  [amount]
+  (-> amount
+      (utils.hex/normalize-hex)
+      (native-module/hex-to-number)
+      (money/wei->ether)
+      (money/with-precision precision)
+      (str)))
+
+(defn- process-send-activity
+  [{:keys [symbol-out chain-id-out amount-out]} activity chain-id->network-name]
+  (let [network-name (chain-id->network-name chain-id-out)]
+    (assoc activity
+           :transaction  :send
+           :token        symbol-out
+           :amount       (activity-amount amount-out)
+           :network-name network-name
+           :network-logo (quo.resources/get-network network-name))))
+
+(defn- process-receive-activity
+  [{:keys [symbol-in amount-in chain-id-in]} activity chain-id->network-name]
+  (let [network-name (chain-id->network-name chain-id-in)]
+    (assoc activity
+           :transaction  :receive
+           :token        symbol-in
+           :amount       (activity-amount amount-in)
+           :network-name network-name
+           :network-logo (quo.resources/get-network network-name))))
+
+(defn- process-activity-by-type
+  [chain-id->network-name
+   {:keys [activity-type activity-status timestamp sender recipient] :as data}]
+  (let [activity {:relative-date (datetime/timestamp->relative (* timestamp 1000))
+                  :timestamp     timestamp
+                  :status        (constants/wallet-activity-status->name activity-status)
+                  :sender        sender
+                  :recipient     recipient}]
+    (condp = activity-type
+      constants/wallet-activity-type-send
+      (process-send-activity data activity chain-id->network-name)
+
+      constants/wallet-activity-type-receive
+      (process-receive-activity data activity chain-id->network-name)
+
+      nil)))
+
+(rf/reg-sub
+ :wallet/activities-for-current-viewing-account
  :<- [:wallet/all-activities]
  :<- [:wallet/current-viewing-account-address]
- (fn [[activities current-viewing-account-address]]
-   (->> activities
-        (filter (fn [{:keys [sender recipient activity-type]}]
-                  (let [receiving-activity? (= activity-type constants/wallet-activity-type-receive)
-                        relevant-address    (if receiving-activity? recipient sender)]
-                    (= relevant-address current-viewing-account-address))))
-        (distinct)
```

Comment:
Is `distinct` not required, even as a safety check? (there was this [bug ](https://github.com/status-im/status-mobile/pull/19906#issuecomment-2110074516))

Reply:
Previously we were using all accounts transactions and filtering the ones where the address appeared, that's why we needed to use `distinct`.

Now the activities in re-frame are stored by account, so there's no place for duplicates :)